src/anyio/from_thread.py

```diff
@@ -177,7 +179,9 @@ def callback(f: Future) -> None:
 
         try:
             retval = func(*args, **kwargs)
-            if iscoroutine(retval):
+            if iscoroutine(retval) or (
+                isawaitable(retval) and not isinstance(retval, DeprecatedAwaitable)
```

Comment:
I had to special case this to get the tests in anyio itself to pass, but there may be a better way

Reply:
Is the `iscoroutine()` check even needed anymore if `isawaitable()` is used?

Reply:
not really, I checked and all tests still pass without it, I was just playing it safe. I removed it in 22c2ccb